src/ansible_navigator/configuration_subsystem/defs_presentable.py

```diff
@@ -0,0 +1,208 @@
+from typing import List
+from typing import NewType
+from typing import Optional
+from typing import Tuple
+from typing import Type
+from typing import TypeVar
+from typing import Union
+
+from .definitions import CliParameters
+from .definitions import Constants as C
+from .definitions import SettingsEntry
+from .utils import create_settings_file_sample
+
+
+PresentableSettingsEntryValue = Union[bool, Dict, str, List]
+
+TCli = TypeVar("TCli", bound="PresentableCliParameters")
+TEnt = TypeVar("TEnt", bound="PresentableSettingsEntry")
+
+
+@dataclass(frozen=True)
+class PresentableCliParameters:
+    """A settings entry's cli parameters in a presentable structure."""
+
+    NO_LONG_MSG: ClassVar[str] = "No long CLI parameter"
+    NO_SHORT_MSG: ClassVar[str] = "No short CLI parameter"
+
+    long: str = NO_LONG_MSG
+    """The long cli parameter value"""
+    short: str = NO_SHORT_MSG
+    """The short cli parameter value"""
+
+    @classmethod
+    def from_cli_params(
+        cls: Type[TCli],
+        cli_parameters: Optional[CliParameters],
+        name_dashed: str,
+    ) -> TCli:
+        """Create an ``_HRCliParameters`` based on an entry's cli parameters.
+
+        :param cli_parameters: The entry's cli parameters
+        :param name_dashed: The dashed name of the parent settings entry
+        :returns: The instance of self based on an entry's cli parameters
+        """
+        if isinstance(cli_parameters, CliParameters):
+            short = cli_parameters.short or cls.NO_SHORT_MSG
+            long = cli_parameters.long(name_dashed)
+            return cls(long=long, short=short)
+        return cls()
+
+
+@dataclass(frozen=True)
+class PresentableSettingsEntry:
+    # pylint: disable=too-many-instance-attributes
+    """A settings entry in a presentable structure."""
+
+    choices: List
+    """The possible values"""
+    current_settings_file: str
+    """The path to the current settings file"""
+    current_value: PresentableSettingsEntryValue
+    """The current, effective value"""
+    default_value: PresentableSettingsEntryValue
+    """The default value"""
+    default: bool
+    """Indicates if the current value == the default"""
+    description: str
+    """A short description"""
+    env_var: str
+    """The environment variable"""
+    name: str
+    """The name"""
+    settings_file_sample: Union[str, Dict]
+    """A sample settings file snippet"""
+    source: str
+    """The source of the current value"""
+    subcommands: List
+    """A list of subcommands where this entry is available"""
+    cli_parameters: PresentableCliParameters = PresentableCliParameters()
+    """The CLI parameters, long and short"""
+
+    @property
+    def current(self) -> str:
+        """Convert the current value into a string.
+
+        :returns: The current value as a string
+        """
+        return str(self.current_value)
+
+    def get(self, attribute: str):
+        """Allow this dataclass to be treated like a dictionary.
```

Comment:
Except this will throw if the key doesn't exist, and `dict#get` will return `None`.

Reply:
That was intentional, because we shouldn't be getting attributes from a data that don't exist.

Can confirm when the settings PR goes in, but I can't think of anyplace where we require or have a hard coded key/attribute name required in something headed to the UI.

Reply:
(it's also noted in the description below)